### ISSUE TITLE
Switch from mongo-java-driver to mongodb-driver-sync

### ISSUE THREAD

**Issue by pkubowicz** [lukas-krecan/ShedLock#253]
**Is your feature request related to a problem? Please describe.**
When I upgrade to Spring Boot 2.3, I get
```
Caused by: java.lang.NoSuchMethodError: 'java.lang.String com.mongodb.connection.ClusterSettings.getDescription()'
	at com.mongodb.connection.DefaultClusterFactory.createCluster(DefaultClusterFactory.java:179)
	at com.mongodb.client.internal.MongoClientImpl.createCluster(MongoClientImpl.java:190)
	at com.mongodb.client.internal.MongoClientImpl.<init>(MongoClientImpl.java:57)
	at com.mongodb.client.MongoClients.create(MongoClients.java:112)
	at org.springframework.boot.autoconfigure.mongo.MongoClientFactorySupport.createMongoClient(MongoClientFactorySupport.java:62)
```
This is because net.javacrumbs.shedlock:shedlock-provider-mongo:4.10.1 pulls org.mongodb:mongo-java-driver:3.8.2 while org.springframework.data:spring-data-mongodb:3.0.0.RELEASE pulls org.mongodb:mongodb-driver-core:4.0.3

**Describe the solution you'd like**
As described in [Upgrading to the 4.0 Driver](https://mongodb.github.io/mongo-java-driver/4.0/upgrading/):
> The mongo-java-driver and mongodb-driver “uber-jars” are no longer published. Applications that reference either of these artifacts must switch to either mongodb-driver-sync or mongodb-driver-legacy, depending on which API is being used. Care should be taken to ensure that neither of the uber-jars are included via a transitive dependency, as that could introduce versioning conflicts.

You are probably still able to use the old version of Mongo driver, it is published to Maven: https://mvnrepository.com/artifact/org.mongodb/mongodb-driver-sync/3.8.2

**Describe alternatives you've considered**
Since I use Gradle I am able to use a convenient workaround:
```
modules {
    module("org.mongodb:mongo-java-driver") {
        replacedBy("org.mongodb:mongodb-driver-sync", "https://mongodb.github.io/mongo-java-driver/4.0/upgrading/")
    }
}
```
but the out-of-the-box experience of ShedLock is not good.

**Comment by lukas-krecan** [comment]
Thanks for reporting, I will take a look at it.

**Comment by lukas-krecan** [comment]
Released as 4.11.1

**Comment by pkubowicz** [comment]
Thank you!
